app/models/concerns/client_search.rb

```diff
@@ -64,6 +66,7 @@ def self.text_searcher(text, sorted:, resolve_for_join_query: false)
       # dummy condition to start the OR chain. This method needs refactoring
       where ||= never_cond
       where = search_by_external_id(where, text) if alpha_numeric && respond_to?(:search_by_external_id) && RailsDrivers.loaded.include?(:hmis_external_apis) && HmisExternalApis::AcHmis::Mci.enabled?
+      where = scan_card_search(where, text) if alpha_numeric
```

Comment:
Yes we can restrict to HMIS.

What do you mean by dedicated search? The logic flow of this function is so strage.

We don't know anything for certain about the shape of the scan card code, aside from the fact that it's alphanumeric.  I suppose we could check it first before anything else, and short-circuit the search if we get a result. This kind of does the opposite..skips the name-short-circuit if it _is_ a match. I'll try adding it earlier and see if it makes more sense.

Reply:
Ok, changed: https://github.com/greenriver/hmis-warehouse/pull/3923/files#diff-e0530ee1d7a90940cb0b96a903efe907a875c5495aa71f1088a00488cd20606c 
This makes more sense to me

Reply:
Re dedicated search: sorry what I meant was that we could have a much simpler method that only searched by scan code and would return just the one matching client record. That would mean we'd use a dedicated search input. With this approach it looks like we plan to use the existing client search for scan cards instead

Reply:
Oh I see. Yeah we landed on that for now, to reduce the feature scope (no new frontend work needed for enrollment and bed night workflows). It should be fine because the migrated-in codes have format `P123124` (P + number)